code/datums/ai/dog/dog_controller.dm

```diff
@@ -0,0 +1,275 @@
+		blackboard[BB_FETCH_RESET_IGNORE_CD] = world.time
+		blackboard[BB_FETCH_IGNORE_LIST] = list()
+
+	// if we were just ordered to heel, chill out for a bit
+	if((world.time < blackboard[BB_DOG_HEEL_CD] + AI_DOG_HEEL_DURATION))
+		return
+
+	// if we're not already carrying something and we have a fetch target (and we're not already doing something with it), see if we can eat/equip it
+	if(!blackboard[BB_SIMPLE_CARRY_ITEM] && blackboard[BB_FETCH_TARGET])
+		var/atom/movable/interact_target = blackboard[BB_FETCH_TARGET]
+		if(in_range(living_pawn, interact_target) && (isturf(interact_target.loc)))
+			current_movement_target = interact_target
+			if(IS_EDIBLE(interact_target))
+				current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/eat_snack)
+			else
+				current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/simple_equip)
+			return
+
+	// if we're carrying something and we have a destination to deliver it, do that
+	if(blackboard[BB_SIMPLE_CARRY_ITEM] && blackboard[BB_FETCH_DELIVER_TO])
+		var/atom/return_target = blackboard[BB_FETCH_DELIVER_TO]
+		if(!can_see(pawn, return_target, length=AI_DOG_VISION_RANGE))
+			// if the return target isn't in sight, we'll just forget about it and carry the thing around
+			blackboard[BB_FETCH_DELIVER_TO] = null
+			return
+		current_movement_target = return_target
+		current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/deliver_item)
+		return
+
+	// occasionally see if there's any loose snacks in sight nearby
+	if(DT_PROB(40, delta_time))
+		for(var/obj/item/potential_snack in oview(living_pawn,2))
+			if(IS_EDIBLE(potential_snack) && (isturf(potential_snack.loc) || ishuman(potential_snack.loc)))
+				current_movement_target = potential_snack
+				current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/eat_snack)
+				return
+
+/datum/ai_controller/dog/PerformIdleBehavior(delta_time)
+	var/mob/living/living_pawn = pawn
+	if(!isturf(living_pawn.loc) || living_pawn.pulledby)
+		return
+
+	// if we were just ordered to heel, chill out for a bit
+	if((world.time < blackboard[BB_DOG_HEEL_CD] + AI_DOG_HEEL_DURATION))
+		return
+
+	// if we're just ditzing around carrying something, occasionally print a message so people know we have something
+	if(blackboard[BB_SIMPLE_CARRY_ITEM] && DT_PROB(5, delta_time))
+		var/obj/item/carry_item = blackboard[BB_SIMPLE_CARRY_ITEM]
+		living_pawn.visible_message("<span class='notice'>[living_pawn] gently teethes on \the [carry_item] in [living_pawn.p_their()] mouth.</span>", vision_distance=COMBAT_MESSAGE_RANGE)
+
+	if(DT_PROB(5, delta_time) && (living_pawn.mobility_flags & MOBILITY_MOVE))
+		var/move_dir = pick(GLOB.alldirs)
+		living_pawn.Move(get_step(living_pawn, move_dir), move_dir)
+	else if(DT_PROB(10, delta_time))
+		living_pawn.manual_emote(pick("dances around.","chases its tail!"))
+		//INVOKE_ASYNC(GLOBAL_PROC, .proc/dance_rotate, living_pawn)
```

Comment:
? This kills tactical dog spinning?

Reply:
spinning implemented as /datum/component/spinny, though sadly it spins somewhat slower